tests/phpunit/CRM/RcBase/Api/CreateHeadlessTest.php

```diff
@@ -0,0 +1,365 @@
+        CRM_RcBase_Api_Create::email(0, $email);
+    }
+
+    /**
+     * @throws CRM_Core_Exception
+     */
+    public function testCreateEmailWithMissingRequiredFields()
+    {
+        $email = [
+            'location_type_id' => 2,
+        ];
+
+        // Create email
+        $this->expectException(CRM_Core_Exception::class, "Invalid exception class");
+        CRM_RcBase_Api_Create::email($this->testContactId, $email);
+    }
+
+    /**
+     * @throws CRM_Core_Exception
+     */
+    public function testCreatePhone()
+    {
+        $phone = [
+            'phone' => '+12343243',
+            'location_type_id' => 1,
+        ];
+
+        // Create phone
+        $phone_id = CRM_RcBase_Api_Create::phone($this->testContactId, $phone);
+
+        // Get data
+        $id = cv("api4 Phone.get +s id +w phone=".$phone['phone']);
+        $data = cv("api4 Phone.get +s phone,location_type_id +w phone=".$phone['phone']);
+
+        // Check valid ID
+        $this->assertSame($id[0]['id'], $phone_id, 'Bad phone ID returned');
+
+        // Check valid data
+        unset($data[0]['id']);
+        $this->assertSame($data[0], $phone, 'Bad phone data returned');
+
+        // Check invalid ID
+        $this->expectException(CRM_Core_Exception::class, "Invalid exception class");
+        CRM_RcBase_Api_Create::phone(-4, $phone);
+    }
+
+    /**
+     * @throws CRM_Core_Exception
+     */
+    public function testCreateAddress()
+    {
+        $address = [
+            'city' => 'Rome',
+            'location_type_id' => 1,
+        ];
+
+        // Create address
+        $address_id = CRM_RcBase_Api_Create::address($this->testContactId, $address);
+
+        // Get data
+        $id = cv("api4 Address.get +s id +w city=".$address['city']);
+        $data = cv("api4 Address.get +s city,location_type_id +w city=".$address['city']);
+
+        // Check valid ID
+        $this->assertSame($id[0]['id'], $address_id, 'Bad address ID returned');
+
+        // Check valid data
+        unset($data[0]['id']);
+        $this->assertSame($data[0], $address, 'Bad address data returned');
+
+        // Check invalid ID
+        $this->expectException(CRM_Core_Exception::class, "Invalid exception class");
+        CRM_RcBase_Api_Create::address(0, $address);
+    }
+
+    /**
+     * @throws CRM_Core_Exception
+     */
+    public function testCreateRelationship()
+    {
+        $contact_other = [
+            'values' => [
+                'contact_type' => 'Individual',
+                'first_name' => 'Marcus',
+                'last_name' => 'Crassus',
+            ],
+        ];
+        $user_other = cv("api4 Contact.create '".json_encode($contact_other)."'");
+        $contact_id_other = (int)$user_other[0]['id'];
+
+        $relationship = [
+            'contact_id_b' => $contact_id_other,
+            'relationship_type_id' => 1,
+            'description' => 'Test',
+        ];
+
```

Comment:
I think the number of relationships with test description has to be checked here, to prove that the given relationship does not exist before the api call.